### PR TITLE
[MBTA] update locations and routes

### PR DESCRIPTION
**Pull request by ava-silver**
Update locations, add typing, and do a refactor of the `Source` type.

### CODE REVIEW

File: wallmatrix/sources/mbta.py
Comment (breqdev) [inline wallmatrix/sources/mbta.py]: We could replace the orange line with the 1 bus (based on https://api-v3.mbta.com/routes I believe the ID is `line-1`). We should verify that the code works on buses first though :)

File: wallmatrix/sources/mbta.py
Comment (breqdev) [inline wallmatrix/sources/mbta.py]: `walk_min` values need to be updated -- this represents the time it takes to get from our place to the corresponding stop.

File: wallmatrix/sources/mbta.py
Comment (breqdev) [inline wallmatrix/sources/mbta.py]: We may want to put the green line above the other line, since it's the one we take most frequently.

**Review by breqdev — CHANGES_REQUESTED** [review]
LGTM pending the `walk_min` value updates; I'll leave it up to you whether we want to go with the OL or the 1 bus for the other line of data.

File: wallmatrix/sources/mbta.py
Comment (ava-silver) [reply wallmatrix/sources/mbta.py]: true!

**Review by breqdev — APPROVED** [review]
LGTM :shipit:

**Pull request merged**
